realm/realm-library/src/main/java/io/realm/DynamicRealmObject.java

```diff
@@ -775,6 +775,16 @@ public void setObject(String fieldName, @Nullable DynamicRealmObject value) {
 
         switch (columnType) {
             case LIST:
+                // Due to type erasure it is not possible to check the generic parameter,
+                // instead we try to see if the first element is of the wrong type in order
+                // to throw a better error message.
+                // Primitive types are checked inside `setModelList`
+                if (!list.isEmpty()) {
+                    E element = list.first();
```

Comment:
I think just catching the possible `ClassCastException` and re-throwing it with the custom message would be more reliable and less hacky.

I like the message though!

Reply:
The classcast exception would not catch the generic parameter, only if `RealmList` was of the wrong type. Type erasure is a pain :(

Reply:
I see, I guess it'll do. Thank you! 🙂